### ISSUE TITLE
Some Neomake Async Maker Jobs Unable to Spawn Processes on Windows

### ISSUE THREAD

**Issue by zovt** [neovim/neovim#5376]
- `nvim --version`:

```
NVIM v0.1.6-90-g0a0f72b
Build type: Release
Compilation: C:/msys64/mingw64/bin/gcc.exe -Wconversion -O2 -DNDEBUG -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -D__USE_MINGW_ANSI_STDIO -D_WIN32_WINNT=0x0600 -Wvla -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -IC:/Projects/neovim-1kmqj/build/config -IC:/Projects/neovim-1kmqj/src -IC:/Projects/neovim-1kmqj/.deps/usr/include -IC:/Projects/neovim-1kmqj/.deps/usr/include -IC:/Projects/neovim-1kmqj/.deps/usr/include -IC:/msys64/mingw64/include -IC:/msys64/mingw64/include -IC:/Projects/neovim-1kmqj/build/src/nvim/auto -IC:/Projects/neovim-1kmqj/build/include
Compiled by appvyr-winppveyor@APPVYR-WIN

Optional features included (+) or not (-): -acl   +iconv    -jemalloc -tui
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM\sysinit.vim"
  fall-back for $VIM: "C:/Program Files (x86)/nvim/share/nvim"
```
- Vim (version: ) behaves differently? N/A
- Operating system/version: Windows 10
- Terminal name/version: N/A
- `$TERM`: N/A
### Actual behaviour

Some neomake makers, including 'stylelint' and 'csslint' are throwing errors when trying to create the async job.

See here for the errors: https://github.com/neomake/neomake/issues/618#issuecomment-249035917
### Expected behaviour

The makers work without error

**Comment by equalsraf** [comment]
The problematic command was this, right?

> :echo jobstart(['csslint', '--format=compact', '/path/to/css'])

FYI you might need  #5280 for jobstart() to work in windows. But maybe in this case you get away without it.

I did a quick browse in the csslint wiki, and I think the command is actually csslint.bat, but Neovim cannot execute a bat file with `jobstart([])`. But one of the following alternatives should work

1. `system(['cmd', '/c', 'csslint --format=compact /path/to/css'])`
1. `system('csslint --format=compact /path/to/css')`

But the second form might need quoting/escaping, etc.

**Comment by zovt** [comment]
I installed csslint with npm, so `which csslint` shows a file that looks like a shell script to me

**Comment by jamessan** [comment]
> But the second form might need quoting/escaping, etc.

So might the first for the general case, since it's being interpreted by cmd.

**Comment by equalsraf** [comment]
> I installed csslint with npm, so which csslint shows a file that looks like a shell script to me

Its the same. When called with `jobstart([...])` it will only run executable files (.exe, .com). And I'm not sure if we should change this behaviour (https://github.com/neovim/neovim/pull/5280#issuecomment-244877246).

If the command works from cmd.exe, then one of the alternatives I used in the previous comment should work.

**Comment by equalsraf** [comment]
> So might the first for the general case, since it's being interpreted by cmd.

Not sure, since the whole command is being passed as single argument, do we need to escape the shell metacharacters in this case too? Argument quoting is already being done for the first case.

**Comment by gaving** [comment]
Is there any workaround for this? Basically meaning here I can't use Neomake configured with `eslint` on Windows, which is sort of a workflow killer. :face_with_head_bandage:

**Comment by justinmk** [comment]
@gaving On Windows, I suggest using Vim until we release 0.2 (hopefully by end of December).

**Comment by gaving** [comment]
@justinmk Yeah no worries, I've little other complaints and everything is stable enough it's just the odd missing annoyance such as this. I'll continue to hold tight!

**Comment by justinmk** [comment]
Fixed by #5280. See https://github.com/neovim/neovim/pull/5280#issue-174549578 and `:help system()` and `:help jobstart()` for complete description of behavior.
